qiskit_metal/qlibrary/core/qroute.py

```diff
@@ -554,6 +554,29 @@ def make_elements(self, pts: np.ndarray):
                                layer=p.layer,
                                subtract=True)
 
+    def to_script(self) -> str:
```

Comment:
This may help, it is in base.py:
 @classmethod
    def _gather_all_children_options(cls) -> dict:

Reply:
This is because they don't all have the same arguments. 
For example `self.type` is not in the options dictionary for QRoute

I'm not very familiar with the metadata though so I can check if it's in there.

Reply:
@grace-harper-ibm , I think both metadata and options would need to be saved and can be through the base.py class.  I  don't know why creators of subclasses HAVE to write a method called to_script.  Just trying to reduce what a component developer needed to do.